docs/source/control_tutorials.rst

```diff
@@ -355,26 +369,174 @@ As with :py:class:`.MachineController`, :py:class:`.BMPController` supports the
 
-These are used internally by :py:class:`.MachineController` and
-:py:class:`.BMPController`. Users are encouraged to read the official SDP and
-SCP App-Notes and refer to the Rig source code for further guidance in using SCP
-and SDP directly in applications.
+To receive SDP packets from an application there must first be an open socket
+ready to receive the packets. For example::
+
+    >>> import socket
+    >>> PORT = 50007
+    >>> in_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    >>> in_sock.bind(("", PORT))
+
+Next, you must set up an 'IP tag' on every Ethernet-connected SpiNNaker chip
+through which SDP packets may be sent back to the host which informs SpiNNaker
+of the IP address these packets should be sent to.
+
+A list of the Ethernet-connected chips in a typical SpiNNaker machine can be
+produced using :py:class:`rig.machine_control.MachineController.get_machine`
+and :py:class:`rig.geometry.spinn5_eth_coords` and an IP tag configured on each
+using :py:class:`rig.machine_control.MachineController.iptag_set` like so::
+
+    >>> from rig.machine_control import MachineController
+    >>> from rig.geometry import spinn5_eth_coords
+    
+    >>> # Get the IP and port of the socket we opened
+    >>> addr, port = in_sock.getsockname()
+    
+    >>> # Set-up IP Tag 1 on each ethernet-connected chip to forward all SDP
+    >>> # packets to this socket.
+    >>> mc = MachineController("spinnaker-machine-hostname")
+    >>> machine = mc.get_machine()
+    >>> for x, y in spinn5_eth_coords(machine.width, machine.height):
+    ...     mc.iptag_set(1, addr, port, x, y)
+
+You can now listen for incoming packets and unpack them using
+:py:meth:`rig.machine_control.packets.SDPPacket.from_bytestring` and
+:py:meth:`rig.machine_control.packets.SCPPacket.from_bytestring`. For example,
+to unpack SCP packets received from the machine::
+
+    >>> from rig.machine_control.packets import SCPPacket
+    >>> while True:
+    ...     data = self.in_sock.recv(512)
+    ...     if not data:
+    ...         break
+    ...     packet = SCPPacket.from_bytestring(data)
+    ...     print("Got SCP packet from core {packet.src_cpu} "
+    ...           "of chip ({packet.src_x}, {packet.src_y}) "
+    ...           "with cmd_rc {packet.cmd_rc} and data "
+    ...           "{packet.data}.".format(packet=packet))
 
 .. note::
-    Since different applications typically have very different requirements for
-    SDP and SCP support, Rig does not currently offer any high-level support for
-    their use. The developers are open to discussions about potential
-    (appropriate) high-level interfaces.
+
+    We use a 512 byte UDP receive buffer since at present the largest SDP
+    packet supported by the machine at the time of writing is 256 bytes + SDP
+    header. The :py:class:`~rig.machine_control.MachineController`'s
+    :py:meth:`~rig.machine_control.MachineController.scp_data_length` property
+    can be used to get the actual value.
```

Comment:
But it only gets the length of the data; perhaps we should note what the header size is and repeat the argument that you should use the smallest power of two larger than that for performance reasons?